builder/uploader/jenkins_upload.py

```diff
@@ -1,16 +1,37 @@
 import requests
+from functools import partial
 
 headers = {'Content-type': 'text/xml'}
 
 
+def krb_request(request_fn, config):
```

Comment:
I might be too pedantic, but wouldn't it be better to send `auth_fallback` which will contain `(user, password)` tuple, instead of the config? Leaving up to you..